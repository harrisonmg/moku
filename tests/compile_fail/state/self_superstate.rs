use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod machine {}

    struct Top;
    impl TopState<BlinkyState> for Top {}

    struct Bottom;

    #[superstate(Bottom)]
    impl State<BlinkyState> for Bottom {}
}

fn main() {}
