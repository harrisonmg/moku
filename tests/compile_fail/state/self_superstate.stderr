error: superstate `Bottom` doesn't match any other known `moku::State` or `moku::TopState`
  --> tests/compile_fail/state/self_superstate.rs:15:5
   |
15 |     #[superstate(Bottom)]
   |     ^
